src/please.go

```diff
@@ -1038,30 +1038,33 @@ func initBuild(args []string) string {
 			log.Warning("%s", http.ListenAndServe(fmt.Sprintf("127.0.0.1:%d", opts.ProfilePort), nil))
 		}()
 	}
+
+	return command
+}
+
+func main() {
+	command := initBuild(os.Args)
+
+	var cpuFile *os.File
```

Comment:
I wonder if it would be neater to kick the whole lot into another function so you can defer things in there, and have it return an exit code that `main` can pass to `os.Exit`. Would be nicer to be able to use `defer` rather than dancing around the potential `os.Exit`.

Reply:
oh yeah, how did this ever think it would work?